test/apps/iodemo/io_demo.cc

```diff
@@ -2029,6 +2079,10 @@ class DemoClient : public P2pDemoCommon {
         assert(active_index < _num_active_servers_to_use);
         --_num_active_servers_to_use;
         active_servers_swap(active_index, _num_active_servers_to_use);
+
+        if (_next_active_index == _num_active_servers_to_use) {
+            _next_active_index = 0;
+        }
```

Comment:
IIUC correctly this is `active_index == _next_active_index` and it's last one, IMHO can be clearer to update index before remove/swap like this:
```
if (active_index == _next_active_index) {
    get_next_active_index();
}
```
also this way `_next_active_index` is changed by only 1 function

Reply:
fixed in another way to cover case when last active server isn't visited during IO

Reply:
better to change iterators (`_next_active_index`) only one way - `get_next_active_index()` in this case

Reply:
> better to change iterators (`_next_active_index`) only one way - `get_next_active_index()` in this case

we could not use `get_next_active_index()` here

Reply:
see example above

Reply:
> see example above

I meant that we should fix it in another way, See updated code.
Now we do another `swap()` to not skip last elemtn in the active servers if `active_index < _next_active_index`